### PR TITLE
Ensure `toggleLoading()` respects `isLoadingEnabled()`

### LINKED ISSUE #14972: `toggleLoading()` does not honor `isLoadingEnabled()`
Loading indicators should only be allowed for some elements as can be seen [here](https://github.com/ampproject/amphtml/blob/master/src/layout.js#L298) and [here](https://github.com/ampproject/amphtml/blob/master/src/layout.js#L98) in layout.js

However, we are currently forcing all elements to allow the loading indicator as can be seen [here](https://github.com/ampproject/amphtml/blob/master/src/base-element.js#L728) in base-element.js

This is is a bug. This is why we see the loading indicator for[ facebook-* elements](https://ampbyexample.com/components/amp-facebook/) even though they aren't whitelisted.

cc @jridgewell

### PR DESCRIPTION
**Pull request by nainar**
This makes sure `toggleLoading()` isn't forcefully adding loaders to all elements. 

Fixes #14972.

Also adds tests to make sure this doesn't happen again.

### CODE REVIEW

**Comment by nainar** [comment]
@jridgewell could you take a look? My 🕷 senses tell me I am missing a test case.

File: test/functional/test-layout.js [hunk 1]
```diff
@@ -36,6 +36,47 @@ describe('Layout', () => {
...
+      'AMP-PLAYBUZZ',
+      'AMP-VIDEO',
+      'AMP-YOUTUBE',
+    ];
+    elementsValidTagNames.forEach(function(tag) {
+      el.tagName = tag;
+      expect(isLoadingAllowed(el)).to.be.true;
+    });
+
+    // This isn't an exhaustest list of elements that aren't allowed
```
[the remark this answers is not in the document]
Comment (nainar) [reply test/functional/test-layout.js]: I have no excuse :P fixed in the latest commit.

File: test/functional/test-custom-element.js [hunk 2]
```diff
@@ -1643,11 +1643,28 @@ describes.realWin('CustomElement', {amp: true}, env => {
     it('should be enabled by default', () => {
       stubInA4A(false);
       expect(element.isLoadingEnabled_()).to.be.true;
+      // If loading is enabled, we should be able to toggle the classes
+      element.toggleLoading(true);
```
[the remark this answers is not in the document]
Comment (nainar) [reply test/functional/test-custom-element.js]: element is a TestElement which extends BaseElement - so shouldn't this be calling `BaseElement::toggleLoading`?

Debugging

**Comment by aghassemi** [comment]
@nainar @jridgewell this breaks the behaviour introduced by #13782, let's revet before the Canary is cut today and find a different solution here.

**Pull request merged**